wdl-lint/src/v1/mixed_indentation.rs

```diff
@@ -0,0 +1,116 @@
+//! A lint rule for checking mixed indentation in command text.
+
+use wdl_ast::experimental::v1::CommandPart;
+use wdl_ast::experimental::v1::CommandSection;
+use wdl_ast::experimental::v1::Visitor;
+use wdl_ast::experimental::AstToken;
+use wdl_ast::experimental::Diagnostic;
+use wdl_ast::experimental::Diagnostics;
+use wdl_ast::experimental::Span;
+use wdl_ast::experimental::VisitReason;
+
+use super::Rule;
+use crate::util::lines_with_offset;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the mixed indentation rule.
+const ID: &str = "MixedIndentation";
```

Comment:
Q: eventually we want to check for mixed indentation through the whole document, but that is relatively inconsequential when it's not inside the command text (just super ugly and weird, but harmless). Will that be implemented under this same rule? I don't think so, because the `explanation()` will have to be different. Or at least significantly rephrased and extended.

If my hunch is correct and we want command-section-`MixedIndentation` _and_ a **non**-command-section-`MixedIndentation`, naming will be awkward. Should this be renamed `CommandSectionMixedIndentation`? It's wordy. What would the "everywhere else" name be ruled? 🤔

Reply:
> I would favor a single rule that reports slightly different messages. Within the `command` block a `warning` and outside it a `note`.

I was thinking the same until I tried to come up with a decent `explanation`. Imagine someone gets the `note` version due to their meta section. (Somehow) it's not clear to them what's wrong so they follow the `sprocket explain MixedIndentation` instruction. Anything about the command section and its word stripping is irrelevant to that user.

IMO the best user experience will be with 2 different Rules that will be able to communicate the issue most clearly.

> Otherwise, we could do something simple like `NonCommandSectionMixedIndentation` and `CommandSectionMixedIndentation`. It's a bit wordy, but probably the simplest.

Yea I don't love this but IDK any better phrasing for it. In the absence of other suggestions, I think the `ID` for this should become `CommandSectionMixedIndentation`.

Reply:
Sounds like we should rename this rule to `CommandSectionMixedIndentation` and follow-up later with a `MixedIndentation` rule that looks at the document's whitespace instead (and logs a note diagnostic).

Perhaps we should also tag this rule with correctness to justify it using a warning level whereas the new `MixedIndentation` rule isn't about correctness, just style/spacing, and therefore logs a note?

Reply:
All sounds good to me @peterhuene . I like calling _the other_ rule just `MixedIndentation`. That's better than `NonCommand...`. Agree to add `Correctness`. And why don't we remove `Style`?

Reply:
I've added a `Correctness` tag and removed `Style` and renamed the rule to `CommandSectionMixedIndentation`.

We can follow-up with a less-severe `Style` tagged rule named `MixedIndentation` that looks only at document whitespace in a future PR.